### PR TITLE
imp API Triggered Notifications

### PR DESCRIPTION
**Pull request by xxcxy**
Slack notifications screenshots:
[Job Candidate is Selected] https://monosnap.com/file/cd8JqhNb5NthQzVxN3aX38CFa1NQic 
[Overlapping Interview Invites] https://monosnap.com/file/Dm74Kzf50AW40OZOZTU19f79Hj89jl

ps: if you have a slack `webhooks` url, you can set `slackURL` in `scripts/demo-email-notifications/index.js`.

### CODE REVIEW

File: scripts/demo-email-notifications/index.js [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
+const axios = require('axios')
+const _ = require('lodash')
 const moment = require('moment')
 const handlebars = require('handlebars')
 const logger = require('../../src/common/logger')
 const { Interview, JobCandidate, ResourceBooking } = require('../../src/models')
 const { Interviews } = require('../../app-constants')
 
 const consumer = new Kafka.GroupConsumer({ connectionString: process.env.KAFKA_URL, groupId: 'test-render-email' })
+const slackURL = null
```
Comment (maxceem) [inline scripts/demo-email-notifications/index.js:13]: Can we please, config `slackURL` using ENV variable for this script, and update README of this script accordingly.

File: src/eventHandlers/InterviewEventHandler.js [hunk 2]
```diff
@@ -33,6 +37,123 @@ async function sendInvitationEmail (payload) {
...
+            [Op.lt]: interview.startTimestamp
+          }
+        }, {
+          endTimestamp: {
+            [Op.gt]: interview.endTimestamp
+          }
+        }]
+      }]
+    }
+  })
```
Comment (maxceem) [inline src/eventHandlers/InterviewEventHandler.js:72]: We should **not** take into account interviews in status `Completed` or `Cancelled` for both `interview` and `overlappingInterview`.

File: src/eventHandlers/ResourceBookingEventHandler.js [hunk 3]
```diff
@@ -67,6 +68,50 @@ async function placeJobCandidate (payload) {
...
+    }
+  }
+  await helper.postEvent(config.NOTIFICATIONS_CREATE_TOPIC, {
+    notifications: [emailData, webData]
+  })
+  logger.debug({
+    component: 'ResourceBookingEventHandler',
+    context: 'placeJobCandidate',
+    message: `send notifications, teamName: ${project.name}, jobTitle: ${job.title}, projectId: ${project.id}, userHandle: ${user.handle}`
+  })
```
Comment (maxceem) [inline src/eventHandlers/ResourceBookingEventHandler.js:114]: Let's put this code into a separate function.
We create separate functions for each separate logic. For example method `placeJobCandidate` is to change status of JobCandidate when RB is placed.

File: src/eventHandlers/JobEventHandler.js [hunk 4]
```diff
@@ -65,6 +67,55 @@ async function processUpdate (payload) {
   await cancelJob(payload)
 }
 
+/**
+ * Process job create event.
+ *
+ * @param {Object} payload the event payload
+ * @returns {undefined}
+ */
+async function processCreate (payload) {
```
Comment (maxceem) [inline src/eventHandlers/JobEventHandler.js:76]: Can we please, create a separate function to send notifications and call it inside `processCreate` like we do in other places. This way we separate functionality for each kind of logic.

**Review by maxceem — CHANGES_REQUESTED** [review]
Thanks @xxcxy, it works great. Just would like make some code and format improvements.

Please, check comments below for code improvements.

And here are some email/slack format improvements, basicaly:
- update title
- make all the team and job names as links
- remove separate links to keep space
- format dates in EDT timezone using the next date format: `Jan 17, 2020, 7:12 am EDT`

1. New Job Created (email)

    image

2. New Team Created (email)

    image

3. Overlapping Interview Invites (email)

    image
    
4. Job Candidate is Selected (email)
    
    image

5. Resource Booking is Placed (email)

    image

Similar for Slack:
- make team name and job title as links
- remove separate links
- format dates in EDT timezone using the next date format: `Jan 17, 2020, 7:12 am EDT`

image

File: src/common/helper.js [hunk 5]
```diff
@@ -1914,7 +1914,13 @@ async function getTags (description) {
...
  */
 async function createProject (currentUser, data) {
-  const token = currentUser.jwtToken
+  let token
+  if (currentUser.hasManagePermission || currentUser.isMachine) {
+    const m2mToken = await getM2MToken()
+    token = `Bearer ${m2mToken}`
+  } else {
+    token = currentUser.jwtToken
+  }
```
[the remark this answers is not in the document]
Comment (xxcxy) [reply src/common/helper.js:1923]: emmm...  if use the user token, creating team will fail from postman. Will revert this.

[the remark this answers is not in the document]
Comment (maxceem) [reply src/common/helper.js:1923]: got it, but we have to revert it.

to test it from postman locally, you would have to use real token from DEV environment and make the next change to bypass token validation:
```
In tc-project-service/node_modules/tc-core-library-js/lib/auth/verifier.js add this to line 23:

callback(undefined, decodedToken.payload);
return;
```

**Review by maxceem — APPROVED** [review]
Thank you @xxcxy. Works good as per local testing.

**Pull request merged**